src/mono/wasm/testassets/WasmBasicTestApp/App/wwwroot/main.js

```diff
@@ -80,11 +84,14 @@ switch (testCase) {
         break;
 }
 
-const { getAssemblyExports, getConfig, INTERNAL } = await dotnet.create();
+const { setModuleImports, getAssemblyExports, getConfig, INTERNAL } = await dotnet.create();
 const config = getConfig();
 const exports = await getAssemblyExports(config.mainAssemblyName);
 const assemblyExtension = config.resources.assembly['System.Private.CoreLib.wasm'] !== undefined ? ".wasm" : ".dll";
 
+setModuleImports("main.js", {
```

Comment:
this is strange pattern. I would prefer if there was `async Task Main()` which would keep the application running until the test finished. You can use `TaskCompletionSource` to orchestrate that in the `Main()`. 

Then we could just `runtime.runMainAndExit()` and drop `exitWithSuccess`

Reply:
Perhaps that could also allow us to move more logic to C# and drop `JSExport`s too